src/tests/ascent/rover_test_utils.hpp

```diff
@@ -304,25 +346,60 @@ remove_rover_test_data(const std::string &path,
     remove_test_file(output_file);
 }
 
+//-----------------------------------------------------------------------------
+inline void
+get_default_extract_params(Node &extracts,
+                           const std::string &absorption,
+                           const std::string &query_path)
+{
+    extracts["e1/type"] = "xray";
+    extracts["e1/params/rover/absorption"] = absorption;
+    extracts["e1/params/rover/filename"] = query_path;
+    extracts["e1/params/rover/output_type"] = "yaml";
+}
+
 //-----------------------------------------------------------------------------
 inline const bool
 is_vtkm_disabled()
 {
     Node about_info;
     ascent::about(about_info);
-    const bool disabled = "disabled" == about_info["runtimes/ascent/vtkm/status"].as_string();
+    if ("disabled" != about_info["runtimes/ascent/vtkm/status"].as_string())
+    {
+        return false;
+    }
 
     int par_rank = 0;
 #ifdef ASCENT_MPI_ENABLED
     MPI_Comm_rank(COMM, &par_rank);
 #endif
 
-    if (0 == par_rank && disabled)
+    if (0 == par_rank)
+    {
+        ASCENT_INFO("Skipping test: Ascent was built without vtkm\n");
+    }
+
+    return true;
+}
+
+//-----------------------------------------------------------------------------
+#ifdef ASCENT_MPI_ENABLED
+inline const bool
+has_two_mpi_ranks(int par_rank, int par_size)
+{
+    // The rover + MPI tests will fail if there are not exactly 2 ranks
+    // due to the baselines being generated with 2 ranks
```

Comment:
Why will the tests fail? The same images should be generated no matter how many ranks there are. Are the tests only set up to make sense for two ranks?

Reply:
I see @siramok. That makes sense. Thanks for being so thorough! It doesn't hurt any for the tests to pass with more than 2 ranks.